### DISCUSSION TITLE
How to theme it / change colors

### DISCUSSION THREAD

**Discussion by bernhardh** [wireui/wireui#29] [Q&A]
Hello,

first of all, thanks for this great package. 

I am just wondering, how to theme it. For example, how to change the primary color of buttons from indigo to orange?

**Comment by ph7jack** [comment]
Hello thank you so much !!

At the moment this is not possible, but I really liked the suggestion. I'll think about how to implement this system-wide

**Comment by bernhardh** [comment]
Yeah. Would be a good idea. There could be a light/dark theme switch as well.

Maybe all of this could be achived in the config file (`config/wireui.php`)? At least you could use it for the `View/Components/Button.php`, instead of hard coding the values. Or isn't this the way to go?

**Reply by ph7jack** [reply]
I think about implementing it in a simpler way, without needing to define the color in all components. in a global way

Various components use the same color classes, transitions...

**Reply by bernhardh** [reply]
Sounds great. Looking forward to see the changes :)

**Reply by bernhardh** [reply]
Btw. the same goes with for example labels. If you are using the elements on dark background, you cant read anything on forms:

![image](https://user-images.githubusercontent.com/642292/123395213-3d02a800-d5a0-11eb-9f5b-fb4b328660e2.png)

Currently I am publishing all the files and overwrite it, but then I am not upgrade save with the package. So there should be a possibility to configure the default text color, danger/error color, etc.

**Comment by ph7jack** [comment]
Im working now in customize this colors

My idea is in tailwind colors config
https://tailwindcss.com/docs/customizing-colors

Ex: 
```
module.exports = {
  theme: {
    colors: {
     ...
      primary:{...},
      secondary: {...}
      ...
    }
  }
}
```

So anyone can set the default colors for their project.
Ps: I'm also working to have dark mode

**Reply by ph7jack** [reply]
It is now released

**Reply by ph7jack** [reply]
Now is possible to customize colors. I'm planning the Dark Theme

**Reply by bernhardh** [reply]
Thanks! That sound great. I tried it, but now I get on `npm run watch` this error:


```
> mix watch


✖ Mix
  Compiled with some errors in 2.91s

ERROR in ./resources/css/app.css
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
SyntaxError

(3:74) /home/www/resources/css/app.css The `bg-secondary-200` class does not exist. If you're sure that `bg-secondary-200` exists, make sure that any `@import` statements are being properly processed before Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.

  1 | /* ./resources/css/app.css */
  2 | @tailwind base;
> 3 | @tailwind components;
    |                                          ^
  4 | @tailwind utilities;
  5 | 

    at processResult (/home/www/node_modules/webpack/lib/NormalModule.js:701:19)
    at /home/www/node_modules/webpack/lib/NormalModule.js:807:5
    at /home/www/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /home/www/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (/home/www/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at Object.loader (/home/www/node_modules/postcss-loader/dist/index.js:140:7)

1 ERROR in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
webpack compiled with 2 errors
```

I had already the theme colors set:

```
    theme: {
        extend: {
            colors: {
                primary: '#E9C46A',
                secondary: '#495867',
                whiteish: '#F7F7FF'
            },
        },
    },
```


What I am doing wrong?

**Reply by ph7jack** [reply]
the wireui colors must be an object with color tonalities, like tailwind colors
```
{
   100: '#FFF',
   200: '#000',
   ...
}
```

you can use shades generator like https://javisperez.github.io/tailwindcolorshades to create color tonalities

**Comment by ph7jack** [comment]
@bernhardh 
The dark mode is now released.

**Comment by cbaconnier** [comment]
Wouldn't be easier (for us) if the components classes were extracted to a dedicated css file using [`@apply`](https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply) ?
Currently I find difficult to tweak the css. 

- You either need to publish the resources and loose the benefits of updates
- Target the specific element to override the default classes using selector `div span span ...`
- Offering a customization [through the component Class](https://github.com/wireui/wireui/blob/main/src/View/Components/Input.php#L85-L113) but has also its limits. (I'm mostly thinking of spacing, or JS templates)
